bundles/org.openhab.core/src/main/java/org/openhab/core/util/ColorUtil.java

```diff
@@ -211,23 +280,69 @@ public static double[] hsbToXY(HSBType hsb, Gamut gamut) {
      * Transform <a href="https://en.wikipedia.org/wiki/SRGB">sRGB</a> color format to
      * <a href="https://en.wikipedia.org/wiki/HSL_and_HSV">HSV</a> based {@link HSBType}.
      *
-     * @param rgb array of three int with the RGB values in the range 0 to 255.
+     * @param rgbw array of three or four int with the RGB(W) values in the range 0 to 255.
      * @return the corresponding {@link HSBType}.
      * @throws IllegalArgumentException when input array has wrong size or exceeds allowed value range.
      */
-    public static HSBType rgbToHsb(int[] rgb) throws IllegalArgumentException {
-        if (rgb.length != 3 || !inByteRange(rgb[0]) || !inByteRange(rgb[1]) || !inByteRange(rgb[2])) {
+    public static HSBType rgbToHsb(int[] rgbw) throws IllegalArgumentException {
+        if (rgbw.length == 4) {
+            return rgbwToHsb(rgbw);
+        }
+        if (rgbw.length != 3 || !inByteRange(rgbw[0]) || !inByteRange(rgbw[1]) || !inByteRange(rgbw[2])) {
             throw new IllegalArgumentException("RGB array only allows values between 0 and 255");
         }
-        return rgbToHsb(new PercentType[] { convertByteToColorPercent(rgb[0]), convertByteToColorPercent(rgb[1]),
-                convertByteToColorPercent(rgb[2]) });
+        return rgbToHsb(new PercentType[] { convertByteToColorPercent(rgbw[0]), convertByteToColorPercent(rgbw[1]),
+                convertByteToColorPercent(rgbw[2]) });
+    }
+
+    /**
+     * Transform <a href="https://en.wikipedia.org/wiki/HSL_and_HSV">HSV</a> based {@link HSBType} to RGBW.
```

Comment:
I just found this PR and was interested because I think this method could be used instead of doing two passes here:
https://github.com/openhab/openhab-addons/blob/a13fd80bfe5cf0b6c55e13756f42abc27b55796b/bundles/org.openhab.binding.hdpowerview/src/main/java/org/openhab/binding/hdpowerview/internal/handler/HDPowerViewRepeaterHandler.java#L264-L265

Anyway, commenting because this description seems to be copied from the method for the other direction. This one is RGBW to `HSBType`.

Reply:
thanks, included in #3882